<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>949</width>
    <height>452</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QScrollArea" name="scrollArea">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>929</width>
        <height>432</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="2" column="7">
        <widget class="QToolButton" name="debugBtn">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item row="11" column="5">
        <widget class="QSpinBox" name="bufferSbx">
         <property name="maximum">
          <number>1048576</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QLabel" name="label_16">
         <property name="text">
          <string>Cache Directory</string>
         </property>
         <property name="buddy">
          <cstring>cacheLed</cstring>
         </property>
        </widget>
       </item>
       <item row="4" column="4">
        <widget class="QLabel" name="label_18">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Tensorboard Port</string>
         </property>
         <property name="buddy">
          <cstring>tensorboardSbx</cstring>
         </property>
        </widget>
       </item>
       <item row="7" column="1" colspan="2">
        <widget class="QLineEdit" name="trainDeviceLed">
         <property name="text">
          <string>cuda</string>
         </property>
        </widget>
       </item>
       <item row="8" column="5" colspan="3">
        <widget class="QLineEdit" name="deviceIndexesLed"/>
       </item>
       <item row="0" column="3">
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="4" column="0">
        <widget class="QCheckBox" name="exposeTensorboardCbx">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Expose Tensorboard</string>
         </property>
        </widget>
       </item>
       <item row="2" column="4">
        <widget class="QLabel" name="label_17">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Debug Directory</string>
         </property>
         <property name="buddy">
          <cstring>debugLed</cstring>
         </property>
        </widget>
       </item>
       <item row="8" column="4">
        <widget class="QLabel" name="label_22">
         <property name="text">
          <string>Device Indexes</string>
         </property>
         <property name="buddy">
          <cstring>deviceIndexesLed</cstring>
         </property>
        </widget>
       </item>
       <item row="10" column="0">
        <widget class="QLabel" name="label_23">
         <property name="text">
          <string>Gradient Reduce Precision</string>
         </property>
         <property name="buddy">
          <cstring>gradientReduceCmb</cstring>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QCheckBox" name="continueCbx">
         <property name="text">
          <string>Continue From Last Backup</string>
         </property>
        </widget>
       </item>
       <item row="13" column="1">
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>Train Device</string>
         </property>
         <property name="buddy">
          <cstring>trainDeviceLed</cstring>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_20">
         <property name="text">
          <string>Dataloader Threads</string>
         </property>
         <property name="buddy">
          <cstring>dataloaderSbx</cstring>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QCheckBox" name="multiGpuCbx">
         <property name="text">
          <string>Multi-GPU</string>
         </property>
        </widget>
       </item>
       <item row="5" column="5">
        <widget class="QSpinBox" name="validateSbx">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item row="12" column="0">
        <widget class="QLabel" name="label_25">
         <property name="text">
          <string>Temp Device</string>
         </property>
         <property name="buddy">
          <cstring>tempDeviceLed</cstring>
         </property>
        </widget>
       </item>
       <item row="9" column="0">
        <widget class="QCheckBox" name="sequentialCbx">
         <property name="text">
          <string>Sequential Mode Setup</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QCheckBox" name="validateCbx">
         <property name="text">
          <string>Validation</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="workspaceLed"/>
       </item>
       <item row="11" column="4">
        <widget class="QLabel" name="label_24">
         <property name="text">
          <string>Buffer Size (MB)</string>
         </property>
         <property name="buddy">
          <cstring>bufferSbx</cstring>
         </property>
        </widget>
       </item>
       <item row="5" column="4">
        <widget class="QLabel" name="label_19">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Validate After</string>
         </property>
         <property name="buddy">
          <cstring>validateSbx</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="5" colspan="2">
        <widget class="QLineEdit" name="cacheLed"/>
       </item>
       <item row="11" column="0">
        <widget class="QCheckBox" name="asyncGradientCbx">
         <property name="text">
          <string>Async Gradient Reduce</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1" colspan="2">
        <widget class="QSpinBox" name="dataloaderSbx">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="value">
          <number>2</number>
         </property>
        </widget>
       </item>
       <item row="1" column="4" colspan="2">
        <widget class="QCheckBox" name="onlyCacheCbx">
         <property name="text">
          <string>Only Cache</string>
         </property>
        </widget>
       </item>
       <item row="0" column="7">
        <widget class="QToolButton" name="cacheBtn">
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QCheckBox" name="tensorboardCbx">
         <property name="text">
          <string>Tensorboard</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QCheckBox" name="debugCbx">
         <property name="text">
          <string>Debug Mode</string>
         </property>
        </widget>
       </item>
       <item row="12" column="1" colspan="2">
        <widget class="QLineEdit" name="tempDeviceLed"/>
       </item>
       <item row="10" column="1" colspan="2">
        <widget class="QComboBox" name="gradientReduceCmb"/>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Workspace Directory</string>
         </property>
         <property name="buddy">
          <cstring>workspaceLed</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QToolButton" name="workspaceBtn">
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item row="2" column="5" colspan="2">
        <widget class="QLineEdit" name="debugLed">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="5" column="6" colspan="2">
        <widget class="QComboBox" name="validateCmb">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="5" colspan="3">
        <widget class="QSpinBox" name="tensorboardSbx">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="buttonSymbols">
          <enum>QAbstractSpinBox::ButtonSymbols::NoButtons</enum>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>65535</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
         <property name="value">
          <number>6006</number>
         </property>
        </widget>
       </item>
       <item row="10" column="4" colspan="4">
        <widget class="QCheckBox" name="fusedGradientCbx">
         <property name="text">
          <string>Fused Gradient Reduce</string>
         </property>
        </widget>
       </item>
       <item row="3" column="4" colspan="4">
        <widget class="QCheckBox" name="alwaysOnTensorboardCbx">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Always-On Tensorboard</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>scrollArea</tabstop>
  <tabstop>workspaceLed</tabstop>
  <tabstop>workspaceBtn</tabstop>
  <tabstop>cacheLed</tabstop>
  <tabstop>cacheBtn</tabstop>
  <tabstop>continueCbx</tabstop>
  <tabstop>onlyCacheCbx</tabstop>
  <tabstop>debugCbx</tabstop>
  <tabstop>debugLed</tabstop>
  <tabstop>debugBtn</tabstop>
  <tabstop>tensorboardCbx</tabstop>
  <tabstop>alwaysOnTensorboardCbx</tabstop>
  <tabstop>exposeTensorboardCbx</tabstop>
  <tabstop>tensorboardSbx</tabstop>
  <tabstop>validateCbx</tabstop>
  <tabstop>validateSbx</tabstop>
  <tabstop>validateCmb</tabstop>
  <tabstop>dataloaderSbx</tabstop>
  <tabstop>trainDeviceLed</tabstop>
  <tabstop>multiGpuCbx</tabstop>
  <tabstop>deviceIndexesLed</tabstop>
  <tabstop>sequentialCbx</tabstop>
  <tabstop>gradientReduceCmb</tabstop>
  <tabstop>fusedGradientCbx</tabstop>
  <tabstop>asyncGradientCbx</tabstop>
  <tabstop>bufferSbx</tabstop>
  <tabstop>tempDeviceLed</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>tensorboardCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>alwaysOnTensorboardCbx</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>125</y>
    </hint>
    <hint type="destinationlabel">
     <x>600</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tensorboardCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>exposeTensorboardCbx</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>124</y>
    </hint>
    <hint type="destinationlabel">
     <x>114</x>
     <y>157</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tensorboardCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_18</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>124</y>
    </hint>
    <hint type="destinationlabel">
     <x>593</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tensorboardCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>tensorboardSbx</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>33</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>789</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>validateCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_19</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>692</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>validateCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>validateSbx</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>764</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>validateCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>validateCmb</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>40</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>874</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>debugCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_17</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>84</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>648</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>debugCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>debugLed</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>816</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>debugCbx</sender>
   <signal>toggled(bool)</signal>
   <receiver>debugBtn</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>160</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>912</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
